libs/blocks/preflight/accessibility/check-video-captions.js

```diff
@@ -0,0 +1,38 @@
+import { getConfig, getFederatedContentRoot } from '../../../utils/utils.js';
+
+export default async function checkVideoCaptions(elements = [], config = {}) {
+  if (!config?.checks?.includes('video-captions')) return [];
+  const violations = [];
+  const mcpIframes = elements.filter((el) => el.tagName.toLowerCase() === 'iframe' && el.src.includes('video.tv.adobe.com'));
+  const federalRoot = getFederatedContentRoot();
+  const { locale } = getConfig();
+  const geo = (locale?.prefix || '').replace('/langstore', '').replace('/', '');
+  const localesMap = await fetch(`${federalRoot}/federal/preflight/preflight-config.json?sheet=captions-map`)
+    .then((r) => r.json())
+    .then((data) => data.data.map((l) => [l.captions, l.geos.split(',')]));
+  const captionsLang = localesMap.find(([, geos]) => geos.includes(geo))?.[0];
+  for (const iframe of mcpIframes) {
+    const videoId = iframe.src.match(/\/v\/(\d+)/)?.[1];
+    // eslint-disable-next-line
+    if (!videoId) continue;
+    const videoData = await fetch(`https://video.tv.adobe.com/vc/${videoId}/${captionsLang}.json`)
+      .then((res) => res.json());
+    // eslint-disable-next-line
+    if (videoData?.captions?.length > 0) continue;
+    if (captionsLang !== 'eng') {
+      const engCaptions = await fetch(`https://video.tv.adobe.com/vc/${videoId}/eng.json`)
+        .then((res) => res.json());
+      // eslint-disable-next-line
+      if (engCaptions?.captions?.length > 0) continue;
+    }
+    violations.push({
+      description: `Video ${videoId} with title "${iframe.title}" is missing captions`,
```

Comment:
I realized that the messaging seems general to videos, but it's actually just MPC videos that support captions anyway. It might make sense to make this clear in the messaging and maybe some method names as well.